test-framework/junit5-component/src/main/java/io/quarkus/test/component/InterceptorMethodCreator.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.test.component;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import io.quarkus.arc.InterceptorCreator;
+import io.quarkus.arc.SyntheticCreationalContext;
+
+public class InterceptorMethodCreator implements InterceptorCreator {
+
+    static final String CREATE_KEY = "createKey";
+
+    private static final Map<String, Function<SyntheticCreationalContext<?>, InterceptFunction>> createFunctions = new HashMap<>();
+
+    @Override
+    public InterceptFunction create(SyntheticCreationalContext<Object> context) {
+        Object createKey = context.getParams().get(CREATE_KEY);
+        if (createKey != null) {
+            Function<SyntheticCreationalContext<?>, InterceptFunction> createFun = createFunctions.get(createKey.toString());
```

Comment:
I don't think `toString()` here is necessary, `Map.get()` takes an `Object` (and not a `K`).

Reply:
Though I see it's the same in `MockBeanCreator`, where I totally missed it :-)

It's a cosmetic thing, anyway.